ppb/events.py

```diff
@@ -85,6 +89,30 @@ def __event__(self, bag, fire_event):
                     raise
 
 
+class TreeStructurePublisher(EventMixin, IterableABC):
+    """
+    A mixin to publish events to an iterable's children.
+    """
+
+    def __iter__(self):
+        return (x for x in ())
+
+    def __event__(self, bag, fire_event):
+        super().__event__(bag, fire_event)
+        for child in self.publish_targets:
+            child.__event__(bag, fire_event)
+
+    @property
+    def publish_targets(self):
```

Comment:
I'm kinda wondering if this should be a dunder method? I'm kinda feeling like our core mechanisms should be dunders, to avoid reserving names.

Also, making this a property is kinda meaningless. I can't imagine a situation where it's set once and never updated? A generator property in particular is... ooph.